### PR TITLE
Added documentation for new gcp integration feature

### PR DESCRIPTION
**Pull request by noise-kngdm**
## Description
In this PR has been added information on a new feature that's coming out on `4.2.1`. After working with [this issue](https://github.com/wazuh/wazuh/issues/9774), the Google Cloud integration module has been reworked to make it possible to use multi-threading to maximize the number of EPS that the module can fetch and process.
The results of the modifications are the following:

image
image

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [ ] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

### CODE REVIEW

File: source/gcp/index.rst [hunk 1]
```diff
@@ -46,3 +46,12 @@ The data flow between the Wazuh module and Cloud Pub/Sub looks as follows:
 .. thumbnail:: ../images/gcp/gcp-data-flow.png
     :align: center
     :width: 100%
+
+Performance
+-----------
+
+The Wazuh module for GCP Pub/Sub uses a `Python library <https://googleapis.dev/python/pubsub/2.7.1/index.html/>`_ to pull the messages from the topic. Its implementation has design limitations when integrated with the Wazuh module.
+
+To improve the integration performance, it is highly recommended to focus on horizontal escalation, configuring this module for the same subscription on different machines (managers or agents).
```
Comment (CarlosRS9) [inline source/gcp/index.rst:55]:
```suggestion
To improve the integration performance, it is highly recommended to use horizontal scalability, configuring this module for the same subscription on different machines (managers or agents).
```

Comment (CarlosRS9) [inline source/gcp/index.rst:55]:
```suggestion
To improve the integration performance, it is highly recommended to use horizontal scalability, configuring this module for the same subscription on different machines (managers or agents).
```

**Review by CarlosRS9 — CHANGES_REQUESTED** [review]

File: source/gcp/index.rst (hunk 1 above)
Comment (noise-kngdm) [reply source/gcp/index.rst]: Solved mistake in 79cc10a17da0d8a90535cb7ada0dfa97168d8426.

**Review by CarlosRS9 — APPROVED** [review]

**Review by davidjiglesias — APPROVED** [review]

File: source/user-manual/reference/ossec-conf/gcp-pubsub.rst [hunk 2]
```diff
@@ -89,14 +90,32 @@ For example ``<credentials_file>wodles/gcp-pubsub/credentials.json</credentials_
...
 | **Default value**  | 100         |
 +--------------------+-------------+
 | **Allowed values** | Any integer |
 +--------------------+-------------+
 
+.. _num_threads:
+
+num_threads
+^^^^^^^^^^^^
+.. versionadded:: 4.2.1
```
Comment (s-ocando) [inline source/user-manual/reference/ossec-conf/gcp-pubsub.rst:105]:
```suggestion
.. versionadded:: 4.2.2
```

File: source/gcp/prerequisites/dependencies.rst [hunk 3]
```diff
@@ -44,8 +44,16 @@ c) From sources:
 google-cloud-pubsub
 -------------------
 
-`google-cloud-pubsub <https://pypi.org/project/google-cloud-pubsub//>`_ is the official python library supported by Google to manage Google Cloud Pub/Sub resources. It will be used to pull the log messages from the Pub/Sub queue. To install this package, execute the following command:
+`google-cloud-pubsub <https://pypi.org/project/google-cloud-pubsub//>`_ is the official python library supported by Google to manage Google Cloud Pub/Sub resources. It will be used to pull the log messages from the Pub/Sub queue. Depending on the Wazuh version used, it is necessary to install a different ``google-cloud-pubsub`` version.
+
+1. For Wazuh versions greater or equal to ``4.2.1``:
```
Comment (s-ocando) [inline source/gcp/prerequisites/dependencies.rst:49]:
```suggestion
1. For Wazuh versions greater or equal to ``4.2.2``:
```

**Review by s-ocando — APPROVED** [review]
LGTM

**Pull request merged**